Tradução do arquivo runtime-projects.md

thibonsan opened the pull request:
Tradução do arquivo docs/advanced/runtime/runtime-projects.md do inglês para português (Brasil).

thibonsan committed:
Traduzido até Generics code improvements

Tradução do arquivo do inglês para português (Brasil) até Generics code improvements.

thibonsan committed:
Término da tradução

Tradução do arquivo do inglês para português (Brasil) até o final.

monoman commented on docs/advanced/runtime/runtime-projects.md:
```diff
@@ -4,136 +4,136 @@ redirect_from:
+Projetos correntes para o ambiente de execução do Mono
 
-These track the projects for the VM team.
+Trilha de projetos para a equipe VM.
```
Acompanhamento dos projetos da equipe da VM.

monoman commented on docs/advanced/runtime/runtime-projects.md:
```diff
@@ -4,136 +4,136 @@ redirect_from:
 -------------
 
-Work towards the [Generational GC](/docs/advanced/garbage-collector/sgen/).
+Trabalha com respeito ao [GC Geracional](/docs/advanced/garbage-collector/sgen/).
```
Trabalho no [Coletor de Lixo Geracional]

monoman commented on docs/advanced/runtime/runtime-projects.md:
```diff
@@ -4,136 +4,136 @@ redirect_from:
+-   Implementar uma versão multi-thread do coletor principal.
 
-Lead: Paolo
+Liderança: Paolo
```
Líder: Paolo

monoman commented on docs/advanced/runtime/runtime-projects.md:
```diff
@@ -4,136 +4,136 @@ redirect_from:
+Utilizar o código LLVM otimizado como um backend para o JIT do Mono. Para mais informação veja o [Mono LLVM](/docs/advanced/mono-llvm/) page.
 
-Lead: Zoltan
+Liderança: Zoltan
```
Líder: Zoltan

monoman commented on docs/advanced/runtime/runtime-projects.md:
```diff
@@ -4,136 +4,136 @@ redirect_from:
+-   Revisar todo o código no ambiente de execução que faz suposições sobre os objetos SRE, isso deve levar a uma série de simplicação e correção de erros de longa data.
 
-Lead:
+Liderança:
```
Líder: ?

monoman commented on docs/advanced/runtime/runtime-projects.md:
```diff
@@ -4,136 +4,136 @@ redirect_from:
 -----------
 
-### Frameless methods on x86
+### Métdos sem ponteiros em x86
```
Métodos sem "quadro de pilha" em x86

monoman commented on docs/advanced/runtime/runtime-projects.md:
```diff
@@ -4,136 +4,136 @@ redirect_from:
+### Métdos sem ponteiros em x86
 
-On x86 we can avoid using the frame pointer and gaining an extra register.
+Em x86 podemos evitar usar o ponteiro e ganhar um registro extra.
```
Em x86 podemos evitar o uso do ponteiro para quadro de pilha (FP - Frame Pointer) e ganhar um registro extra.

monoman commented on docs/advanced/runtime/runtime-projects.md:
```diff
@@ -4,136 +4,136 @@ redirect_from:
+Em x86 podemos evitar usar o ponteiro e ganhar um registro extra.
 
-Lead: Mark Probst
+Liderança: Mark Probst
```
Líder: Mark Probst

monoman commented on docs/advanced/runtime/runtime-projects.md:
```diff
@@ -4,136 +4,136 @@ redirect_from:
+Liderança: Mark Probst
 
-### Generics code improvements
+### Melhorias do código generics
```
### Melhorias no código genérico

monoman commented on docs/advanced/runtime/runtime-projects.md:
```diff
@@ -4,136 +4,136 @@ redirect_from:
--   Avoid LDADDR for constrained calls on reference types.
--   Intrinsify and remove boxing for methods on System.Object over primitive types. This is important to make generic collections produce code close to the manually crafted equivalent.
+-   Evitar LDADDR para chamadas restritas sobre tipos de referência.
+-   Intensificar e remover encapsulamento para métodos em System.Object sobre tipos primitivos. Isso é importatnte para fazer as coleções genéricas produzirem código o mais próximo possível do feito manualmente.
```
Gerar código intrínsico (otimizado) e remover o "envelopamento" (boxing) de métodos em System.Object sobre tipos primitivos. Isso é importante para fazer as coleções genéricas produzirem código muito próximo do escrito manualmente.

monoman commented on docs/advanced/runtime/runtime-projects.md:
```diff
@@ -4,136 +4,136 @@ redirect_from:
+-   Intensificar e remover encapsulamento para métodos em System.Object sobre tipos primitivos. Isso é importatnte para fazer as coleções genéricas produzirem código o mais próximo possível do feito manualmente.
 
-Console Ports
+Port para Consoles
```
Porte para Consoles de Jogos

monoman commented on docs/advanced/runtime/runtime-projects.md:
```diff
@@ -4,136 +4,136 @@ redirect_from:
+### Mudanças na API
 
-As part of Mono 2.8, we will be breaking the embedding API to clean up many issues that have cropped over the years.
+Como parte do Mono 2.8, As part of Mono 2.8, estaremos quebrando a API embarcada par limpar muitas questões que surgiram ao longo dos anos.
```
Como parte do Mono 2.8, estaremos quebrando a API embarcada para limpar muitos problemas que surgiram ao longo dos anos.

monoman commented on docs/advanced/runtime/runtime-projects.md:
```diff
@@ -4,136 +4,136 @@ redirect_from:
 -----------------
 
-Support for configuring Mono as a cross-compiler. Initially it is OK to mandate that the host and the target have the same endianess and the same word size.
+Suporte para configurar Mono como um compilador multiplataforma.Support for configuring Mono as a cross-compiler. Incialmente está OK para quando a origem e o destino possuem a mesma extremidade e o mesmo tamanho de palavra.
```
Suporte para configurar o Mono para compilação cruzada. Inicialmente está OK para quando a origem e o destino possuem a mesma ordenação de bits (endianess) e o mesmo tamanho de 'palavra' (32/64 bits).

monoman commented on docs/advanced/runtime/runtime-projects.md:
```diff
@@ -4,136 +4,136 @@ redirect_from:
--   Mono.Simd for NEON (ARM processors) and/or VMX (Cell Processor).
--   C++ bindings without C glue.
+-   Mono.Simd para NEON (processadores ARM) e/ou VMX (processador de celular).
+-   Bindings C++ sem C.
```
Ligação ao código C++ sem necessitar código intermediário em C.

monoman commented on docs/advanced/runtime/runtime-projects.md:
```diff
@@ -4,136 +4,136 @@ redirect_from:
--   JIT Regalloc: A new register allocation framework.
--   [SafeHandles](/docs/advanced/safehandles/): Support for 2.0 SafeHandles.
+-   [Linear](/docs/advanced/runtime/docs/linear-ir/): Uma atualização para a representação interna (RI) do JIT.
+-   JIT Regalloc: Um novo framework de alocação de registro.
```
registro -> registros.

monoman commented on docs/advanced/runtime/runtime-projects.md:
```diff
@@ -4,136 +4,136 @@ redirect_from:
+### Melhorias do código generics
 
-Generics IL is a bit different from regular code and the current JIT doesn't properly handle it.
+A linguagem intermediária do generics é um pouco diferente do código usual e o JIT atual não consegue tratá-lo adequadamente.
```
A código IL do código genérico é um pouco diferente do código usual e o JIT atual não consegue tratá-lo adequadamente.

monoman commented on docs/advanced/runtime/runtime-projects.md:
```diff
@@ -4,136 +4,136 @@ redirect_from:
+Liderança: Zoltan
 
-Reflection Fixes
+Correções de Reflexão
```
Correções na Introspecção

monoman commented on docs/advanced/runtime/runtime-projects.md:
```diff
@@ -4,136 +4,136 @@ redirect_from:
+Suporte para tipos definidos pelo usuário (utilizado pelo PostSharp e outras ferramentas similares).
 
-This is a very time intensive task as it requires to test all entry points in corlib that take System.Type or System.Reflection.\* types.
+Esta é uma tarefa intensiva que consome muito tempo, pois requer testar todos os pontos de entrada na corlib que levam os tipos System.Type ou System.Reflection.\*.
```
Esta é uma tarefa que consome muito tempo, pois requer testar todos os pontos de entrada na corlib que aceitam os tipos System.Type ou System.Reflection.

monoman commented:
Favor efetuar as correções. Grato pelo esforço :+1:

thibonsan committed:
Correções da tradução

Correções efetuadas na tradução do arquivo que não foram feitas ou estavam incorretas.

thibonsan commented:
Correções efetuadas...
